packetbeat/protos/dns/dns.go

```diff
@@ -707,3 +714,160 @@ func decodeDnsPacket(data []byte) (dns *layers.DNS, err error) {
 	}
 	return d, nil
 }
+
+// TCP implementation
+
+func (dns *Dns) Parse(pkt *protos.Packet, tcpTuple *common.TcpTuple, dir uint8, private protos.ProtocolData) protos.ProtocolData {
+	defer logp.Recover("DNS ParseTcp")
+
+	logp.Debug("dns", "Parsing packet addressed with %s of length %d.",
+		pkt.Tuple.String(), len(pkt.Payload))
+
+	priv := dnsPrivateData{}
+
+	if private != nil {
+		var ok bool
+		priv, ok = private.(dnsPrivateData)
+		if !ok {
+			priv = dnsPrivateData{}
+		}
+	}
+
+	payload := make([]byte, 0)
+
+	// Offset is critical
+	if len(pkt.Payload) > 2 {
+		payload = pkt.Payload[2:]
+	}
+
+	if priv.Data[dir] == nil {
+		priv.Data[dir] = &DnsStream{
+			tcpTuple: tcpTuple,
+			data:     payload,
+			message:  &DnsMessage{Ts: pkt.Ts, Tuple: pkt.Tuple},
+		}
+
+	} else {
+		priv.Data[dir].data = append(priv.Data[dir].data, payload...)
+		if len(priv.Data[dir].data) > tcp.TCP_MAX_DATA_IN_STREAM {
+			logp.Debug("dns", "Stream data too large, dropping DNS stream")
+			priv.Data[dir] = nil
+			return priv
+		}
+	}
+
+	stream := priv.Data[dir]
+	if stream.message == nil {
+		stream.message = &DnsMessage{Ts: pkt.Ts, Tuple: pkt.Tuple}
+	}
+
+	// what kind of checks should be done here ?
```

Comment:
I think the onus is on the parser (gopacket) so no other checks are required. This comment can be removed.